packages/predicate/src/predicate.ts

```diff
@@ -116,11 +117,7 @@ export class Predicate<ARGS extends InputValue[]> extends Account implements Abs
     const paddedCode = new ByteArrayCoder(this.bytes.length).encode(this.bytes);
 
     const OFFSET =
-      VM_TX_MEMORY +
-      TRANSACTION_SCRIPT_FIXED_SIZE +
-      TRANSACTION_PREDICATE_COIN_FIXED_SIZE +
-      paddedCode.byteLength -
-      17;
+      VM_TX_MEMORY + SCRIPT_FIXED_SIZE + INPUT_COIN_FIXED_SIZE + WORD_SIZE + paddedCode.byteLength;
```

Comment:
`-17` was a red herring here. In the `packages/fuel-gauge/src/predicate/predicate-arguments.test.ts`, this test was passing once the `predicateGasUsed` fixed size was added to `INPUT_COIN_FIXED_SIZE`. However, only because `-17` was pointing to a previous byte that also referenced the expected number of `43`, which was pretty lucky. Rather than the required `+1` byte that was needed to correctly offset to `predicateData` - deduced this by using the `chunkByBytes()` helper and affirmed by the fact that the `packages/fuel-gauge/src/vector-types.test.ts` is also now passing (with some changes to the sway predicate to ensure that the tx could actually validate).

However, now that both these tests are passing, and the fixed sizes seem to match that of the Rust SDK (VM TX memory = 10240, script fixed size = 104, Input coin fixed size = 176 + predicate length + predicate data length), I'm having trouble deducing what the additional byte relates to? 🤔 If anyone has time to double check my numbers against the specs when reviewing that would be really helpful.

Reply:
Thanks for checking @camsjams, and yes that was the one I was alluding to. I'm also happy with that outcome so will resolve the conversation.